### DISCUSSION TITLE
Any plan to add the feature to move the mouse with the window

### DISCUSSION THREAD

**Discussion by Dirtybluer** [rxhanson/Rectangle#505] [Ideas]
Every time when I move one window from monitor A to monitor B, the mouse doesn't move along with the window and stays at monitor A. Thus I always need to move the mouse across the monitors afterwards and I find it a bit troublesome.

Would you consider adding a feature that moves the mouse along with the window when move the window among multiple monitors? It could be an option up to users whether to enable this feature. And I think it would be quite helpful.

Also, moving the mouse alone across the multiple monitors and focusing on the top window by default is another useful feature that I have come up with.

At last, many thanks for creating such a decent application.

**Comment by rxhanson** [comment]
Makes sense to me, and it doesn't seem like it would be too hard to implement. 

Where would you expect the cursor to appear? In the top left corner of the window, in the center of the window, etc?

**Reply by Dirtybluer** [reply]
Setting the cursor right in the center in the window would work just fine for me. And it may be nice to leave this configurable to users.

**Comment by alexrobbins** [comment]
It looks like you implemented this for windows that moved to new screens (Works great!). Any chance you could add a flag to enable it for movements within a ~window~ screen, too?

**Reply by rxhanson** [reply]
Sure, I will probably just add it in as a terminal command configuration, though.

**Reply by bnaman50** [reply]
Hey, @rxhanson , 

This is the exact feature I have been looking for such a long time. I was wondering if you have implemented this - 

> Sure, I will probably just add it in as a terminal command configuration, though.

I checked in the [TerminalsCommand](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md) page, but could not find it.  

Could you please let me know how to enable it?

Thanks, 
Naman

**Reply by Aaron-Rumpler** [reply]
`defaults write com.knollsoft.Rectangle moveCursor -int 1` should do it.

**Reply by rxhanson** [reply]
Thanks @Aaron-Rumpler! 

Apologies for forgetting to put this in the TerminalCommands.md. It's now at the bottom:

https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#move-cursor-with-window

**Comment by imagine** [comment]
In practice, you might be shoving a window to the other display to get it out of the way, not to interact with it. A shortcut for snapping the cursor to the center of the active window, however, would be handy.

**Reply by rxhanson** [reply]
I like this idea, and I've also had the request for just moving the cursor to a different display via keyboard shortcut. I don't have the bandwidth to roll it in at the moment, but see no reason to turn down a good pull request for it. If anyone wants to work on it, note that I only accept pull requests that don't modify the UI initially, so it would have to be a terminal command config. With enough demand, I would then consider moving it into the UI.
